src/generic_test.rs

```diff
@@ -25,6 +26,7 @@ impl From<usize> for Data {
     }
 }
 
+#[derive(Clone, Debug, Deserialize, Serialize, Eq, Hash, PartialEq, PartialOrd, Ord)]
```

Comment:
TestPeer is not going to be serialized/deserialized, why these derived traits are added, they seems are not needed?

Reply:
If the instances are derivable, then it is advisable to define them following the [API convention](https://rust-lang-nursery.github.io/api-guidelines/interoperability.html#c-common-traits).

Reply:
It is not API at all!

Reply:
It is a public type in a library. So it is possible of use it outside the library. So technically it **is** part of the library interface. You mat want to change the visibility level if you don't intended that.

Reply:
It is not intended to be used outside library. If you can make it private, you're welcome with another PR. Otherwise these derived traits are not needed.

Reply:
I'll make that in this PR.